File: library/src/main/java/renetik/java/common/CSPoint.java
package renetik.java.common;

public class CSPoint {

	public boolean equals(CSPoint point) {
		return x() == point.x() && y() == point.y();
	}

	@Override public boolean equals(Object o) {
		if (o instanceof CSPoint) return equals((CSPoint) o);
		return super.equals(o);
	}

	@Override public String toString() {
		return "[" + x + ";" + y + "]";
	}

	public final Number x;
	public final Number y;

	public CSPoint(Number x, Number y) {
		this.x = x;
		this.y = y;
	}

	public double doubleX() {
		return x.doubleValue();
	}

	public double doubleY() {
		return y.doubleValue();
	}

	public float x() {
		return x.floatValue();
	}

	public float y() {
		return y.floatValue();
	}

}

File: library/base/src/main/java/renetik/java/math/CSPoint.kt
package renetik.java.math

class CSPoint(val x: Number, val y: Number) {

    fun doubleX() = x.toDouble()
    fun doubleY() = y.toDouble()
    fun x() = x.toFloat()
    fun y() = y.toFloat()
    override fun toString() = "[$x;$y]"

    fun equals(point: CSPoint) = x() == point.x() && y() == point.y()

    override fun equals(other: Any?) = if (other is CSPoint) equals(other) else super.equals(other)
}
